aemedge/scripts/utils.js

```diff
@@ -120,13 +123,31 @@ export function createOptimizedFireflyPicture(
   return picture;
 }
 
+export async function getAccessToken() {
+  const { loadIms } = await import(`${getLibs()}/utils/utils.js`);
+  let authToken;
+  if (!window.adobeIMS) {
+    loadIms().then(async () => {
+      authToken = window.adobeIMS.isSignedInUser() ? window.adobeIMS.getAccessToken().token : null;
+    }).catch(() => {
+      authToken = null;
+    });
+  } else {
+    authToken = window.adobeIMS.isSignedInUser() ? window.adobeIMS.getAccessToken().token : null;
+  }
+  return authToken;
+}
+
 /**
  * Retrieves an array of features from the server.
  * @returns {Promise<Array>} A promise that resolves to an array of features.
  */
-export async function getFeaturesArray(authToken) {
+export async function getFeaturesArray() {
+  const authToken = await getAccessToken();
+  const buildMode = getMetadata('buildmode');
+  const featuresUrl = buildMode === 'stage' ? FEATURES_API_STAGE : FEATURES_API_PROD;
```

Comment:
Question - why do we need buildMode when we can use getEnvironment() that already exists in utils.js?

Reply:
@amol-anand ya..I will get rid of getEnvironment...I think we can use buildmode as it is used for analytics and other places by @kunwarsaluja

Reply:
I would use getEnvironment wherever possible. We should be hitting stage in localhost/.page but prod in .live. Rather than metadata which is part of the content. 

buildmode is ok while developing but shouldn't be relied on long-term. What happens when I want to test something against the stage API in .page but content now has buildmode prod after go-live?

Reply:
fixed